packages/eds-core-react/src/components/next/Input/input.css

```diff
@@ -0,0 +1,120 @@
+    border-color: transparent;
+    outline: 2px solid var(--eds-color-border-focus);
+    outline-offset: -2px;
+  }
+
+  .eds-input--disabled {
+    cursor: not-allowed;
+    color: var(--eds-color-border-medium);
+    border-color: transparent;
+  }
+
+  .eds-input--readonly {
+    border-color: transparent;
+    color: var(--eds-color-text-subtle);
+  }
+
+  /* Invalid state - border color comes from data-color-appearance="danger" */
+  .eds-input--invalid {
+    border-color: var(--eds-color-border-medium);
+  }
+
+  /* Input/Textarea field */
+  .eds-field {
+    width: 100%;
+    border: none;
+    background: transparent;
+    outline: none;
+    color: var(--eds-color-text-strong);
+    font-family: var(--eds-typography-ui-body-font-family), sans-serif;
+    font-size: var(--eds-typography-ui-body-lg-font-size);
+    font-weight: var(--eds-typography-ui-body-lg-font-weight-normal);
+    line-height: var(--eds-typography-ui-body-lg-lineheight-squished);
+    padding-block: var(--eds-selectable-space-vertical);
+    padding-inline: var(--eds-selectable-space-horizontal);
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    resize: none;
+  }
+
+  .eds-field::placeholder {
+    color: var(--eds-color-text-strong);
+  }
+
+  /* 
+   * Disabled field - override browser defaults for consistent styling
+   * Safari applies a default opacity and color to disabled inputs, so we need
+   * to explicitly set opacity: 1 and use -webkit-text-fill-color
+   * Note: We don't rely on inherited color context, so this works consistently
+   */
+  .eds-input--disabled .eds-field {
+    color: var(--eds-color-border-medium);
+    cursor: not-allowed;
+    opacity: 1;
+    -webkit-text-fill-color: var(--eds-color-border-medium);
+  }
+
+  /* Disabled placeholder - maintain same color as disabled text */
+  .eds-input--disabled .eds-field::placeholder {
+    color: var(--eds-color-border-medium);
+    -webkit-text-fill-color: var(--eds-color-border-medium);
+  }
+
+  .eds-input--readonly .eds-field::placeholder {
+    color: var(--eds-color-text-subtle);
+  }
+
+  /* Adornments (icons, text, buttons on left/right) 
+   * Inherits color appearance from parent container
+   * Wrap text in <span data-color-appearance="neutral"> to keep text neutral
+   */
+  .eds-adornment {
+    position: absolute;
+    top: var(--eds-selectable-space-vertical);
+    bottom: var(--eds-selectable-space-vertical);
+    display: flex;
+    align-items: center;
+    color: var(--eds-color-text-subtle);
+    font-family: var(--eds-typography-ui-body-font-family), sans-serif;
+    font-size: var(--eds-typography-ui-body-md-font-size);
+    font-weight: var(--eds-typography-ui-body-md-font-weight-normal);
+    line-height: var(--eds-typography-ui-body-md-lineheight-default);
+  }
+
+  /* Text with neutral appearance should use neutral text color */
+  .eds-adornment [data-color-appearance='neutral'] {
+    color: var(--eds-color-text-subtle);
+  }
```

Comment:
You should apply data-color-appearance in the markup to resolve the neutral color for adornment. This should be the default appearance so it should not be necessary.